### PR TITLE
Feature/new trip mutation

### LINKED ISSUE #34: CSS/JSX: ETA Modal Window
- [ ] Create ETAModal.js component to display user's trip ETA after clicking "Submit Trip" button
- [ ] Complete CSS for modal display (https://www.npmjs.com/package/react-modal)
- [ ] Build sad path functionality for user not wanting to click "Start Trip" button (allow user to click out of modal and return to form)

### LINKED ISSUE #54: New Trip Mutation
1. Write Mutation for a new trip in Form Component

### PR DESCRIPTION
**Pull request by carolineshowscode**
## Type of Change
- [X] JSX
- [ ] HTML
- [ ] JavaScript
- [ ] CSS
- [X] React
- [ ] React Router
- [ ] Cypress
- [X] Apollo
- [X] GraphQL
- [ ] Twilio API
- [ ] Refactor/Fix Bugs

### Feature:
- Defined createTrip mutation - initial issues with the structure and correctly defining the variables within the mutation
- Implemented useMutation hook which creates a mutation function `createTrip` and defines the loading, error and data options
- created one function to fire the `openModal` and `createTrip` functions when the `Submit Trip` button is selected
- within `createTrip` function the variables option was passed to define the value of the required mutation fields
- added conditional rendering for mutationLoading and mutationError properties
- passes mutation result (the `data` property) to TripETA component, this property represents the data returned from our mutation
- created conditional rendering for eta prop passed to `TripETA.js` - without this it will cause React errors because eta prop starts out undefined

## How has this been tested?
- [X] In Browser
- [X] In Console
- [ ] Cypress
- [X] Postman

### CODE REVIEW

File: src/Components/Form/Form.js [hunk 1]
```diff
@@ -31,6 +52,13 @@ function Form({contacts}) {
     setFormattedContacts(formatted);
   }
 
+  function sendTripData() {
+    console.log(endPoint);
+    console.log(startPoint);
+    openModal();
+    createTrip( {variables: {"startPoint": startPoint, "endPoint": endPoint, "travelMode": travelMode.value}});
```
Comment (pcmueller) [inline src/Components/Form/Form.js:57]: Interesting, I see we can use the startPoint and endPoint props as values, straight up, but we need travelMode.value for that one?

File: src/Components/TripETA/TripETA.js [hunk 2]
```diff
@@ -31,12 +31,12 @@ function TripETA( { modalIsOpen, closeModal } ) {
       <div className='eta-modal'>
         <p className='eta-message'>
           <span>Your ETA for this trip is</span>
-          <span>X hour and Y minutes.</span>
+          {eta && <span>{eta.createTrip.trip.eta} minutes</span>}
```
Comment (pcmueller) [inline src/Components/TripETA/TripETA.js:34]: Nice work with this conditional rendering.

File: src/Components/Form/Form.js [hunk 3]
```diff
@@ -3,24 +3,45 @@ import Select from 'react-select';
...
 // import {SearchLocationInput2} from '../SearchLocationInput2/SearchLocationInput2.js'
 // import TripDuration from '../TripDuration/TripDuration';
 
+const CREATE_TRIP = gql `
+  mutation CreateTrip($startPoint: String!, $endPoint: String!, $travelMode: String!){
+    createTrip(input: {startPoint: $startPoint, endPoint: $endPoint, travelMode: $travelMode, userId: 2}) {
+    trip {
+      userId
+      startPoint
+      endPoint
```
Comment (bhohnco) [inline src/Components/Form/Form.js:14]: Nice work here. Easy to understand and readable.

File: src/Components/Form/Form.js [hunk 4]
```diff
@@ -46,19 +74,15 @@ function Form({contacts}) {
   return (
     <form className='trip-form' onSubmit={handleSubmit}>
       <Autocomplete
-          onPlaceSelected={(place,) => {
-            console.log(place);
-          }}
+          onPlaceSelected={(place) => setStartPoint(place.formatted_address)}
```
Comment (bhohnco) [inline src/Components/Form/Form.js:75]: Great add here for the autocomplete!

File: src/Components/Form/Form.js [hunk 5]
```diff
@@ -3,24 +3,45 @@ import Select from 'react-select';
...
 import TripETA from '../TripETA/TripETA';
 import Autocomplete from 'react-google-autocomplete';
-
+import { gql, useMutation } from '@apollo/client';
 // import {SearchLocationInput} from '../SearchLocationInput/SearchLocationInput.js'
 // import {SearchLocationInput2} from '../SearchLocationInput2/SearchLocationInput2.js'
 // import TripDuration from '../TripDuration/TripDuration';
 
+const CREATE_TRIP = gql `
+  mutation CreateTrip($startPoint: String!, $endPoint: String!, $travelMode: String!){
```
Comment (pcmueller) [inline src/Components/Form/Form.js:9]: I'm guessing we can skip propTypes, then, if we're typechecking right in the mutation?

**Review by pcmueller — APPROVED** [review]
Really great job Caroline, thanks for all your work on this.

File: src/Components/TripETA/TripETA.js (hunk 2 above)
Comment (bhohnco) [inline src/Components/TripETA/TripETA.js:34]: nice work getting the time to generate. Looking forward to styling this like the mod 1 days

**Pull request merged**